static/app/utils/profiling/renderers/chartRenderer.tsx

```diff
@@ -110,7 +115,20 @@ export class FlamegraphChartRenderer {
       const origin = vec3.fromValues(0, 0, 1);
       vec3.transformMat3(origin, origin, configViewToPhysicalSpace);
 
-      for (let j = 0; j < serie.points.length; j++) {
+      let start = lowerBound(configView.left, serie.points, a => a.x);
+      let end = upperBound(configView.right, serie.points, a => a.x);
+
+      // Bounds are inclusive, so we adjust start and end by 1. This ensures we
+      // draw the previous/next line that goes outside of bounds.
+      // If we dont do this, the chart looks like | -- | instead of |----|
+      if (start > 0) {
+        start = start - 1;
+      }
+      if (end < serie.points.length) {
+        end = end + 1;
+      }
```

Comment:
do you need to update the checks for `j == 0` and j == series.points.length - 1` to check for start and end -1 here?

Reply:
No, that's really just for when we are fully zoomed out and want to close the draw area - in that case we draw down to x,0. If we don't do that, then we end up with a line from the final (x,y) point to 0,0 which results in a diagonal line across the entire chart.